### PR TITLE
PSP migration script.

### PR DESCRIPTION
**Pull request by jvanz**
Adds a blog post about the script create to migrate PSPs to Kubewarden policies.

### CODE REVIEW

File: content/blog/2022/07/psp-migration-script.md
Comment (jvanz) [inline content/blog/2022/07/psp-migration-script.md:16]: Do not merge this PR until we have the link to the script.

Comment (jvanz) [reply content/blog/2022/07/psp-migration-script.md:16]: https://github.com/kubewarden/utils/pull/1

File: content/blog/2022/07/psp-migration-script.md
Comment (divya-mohan0209) [inline content/blog/2022/07/psp-migration-script.md:2]:
```suggestion
title: Migrate your PSPs to Kubewarden policies!
```

File: content/blog/2022/07/psp-migration-script.md
Comment (divya-mohan0209) [inline content/blog/2022/07/psp-migration-script.md:8]:
```suggestion
As announced in past blog posts, Kubewarden offers 100% coverage for the deprecated,
```

File: content/blog/2022/07/psp-migration-script.md
Comment (divya-mohan0209) [inline content/blog/2022/07/psp-migration-script.md:9]:
```suggestion
and to-be-removed, Kuberernetes PSPs. If everything goes as expected, the PSPs will
```

File: content/blog/2022/07/psp-migration-script.md
Comment (divya-mohan0209) [inline content/blog/2022/07/psp-migration-script.md:10]:
```suggestion
be removed in Kubernetes v1.25 due for release on 23rd August 2022.
```

File: content/blog/2022/07/psp-migration-script.md
Comment (divya-mohan0209) [inline content/blog/2022/07/psp-migration-script.md:11]:
```suggestion
With this in mind, the Kubewarden team has written a script that leverages the
```

File: content/blog/2022/07/psp-migration-script.md
Comment (divya-mohan0209) [inline content/blog/2022/07/psp-migration-script.md:12]:
```suggestion
migration tool written by people from [AppVia](https://github.com/appvia/psp-migration),
```

File: content/blog/2022/07/psp-migration-script.md
Comment (divya-mohan0209) [inline content/blog/2022/07/psp-migration-script.md:17]:
```suggestion
in the Kubewarden Github organization. It will download the
```

File: content/blog/2022/07/psp-migration-script.md
Comment (divya-mohan0209) [inline content/blog/2022/07/psp-migration-script.md:22]:
```suggestion
The script will migrate the PSPs defined in `kubectl` default context.
```

File: content/blog/2022/07/psp-migration-script.md
Comment (divya-mohan0209) [inline content/blog/2022/07/psp-migration-script.md:24]:
```suggestion
apply it directly or save it for further inspection.  Let's take a look at an example:
```

File: content/blog/2022/07/psp-migration-script.md
Comment (divya-mohan0209) [inline content/blog/2022/07/psp-migration-script.md:79]:
```suggestion
Kubewarden installed using the following command:
```

File: content/blog/2022/07/psp-migration-script.md
Comment (divya-mohan0209) [inline content/blog/2022/07/psp-migration-script.md:95]:
```suggestion
If users want to inspect the policies before applying, it's possible to redirect
```

File: content/blog/2022/07/psp-migration-script.md
Comment (divya-mohan0209) [inline content/blog/2022/07/psp-migration-script.md:296]:
```suggestion
The Kubewarden team expects that this will help users migrate from PSPs as soon as possible.
```

File: content/blog/2022/07/psp-migration-script.md
Comment (divya-mohan0209) [inline content/blog/2022/07/psp-migration-script.md:297]:
```suggestion
Let us know if you run into issues. We are happy to help!
```

**Review by divya-mohan0209 — CHANGES_REQUESTED** [review]
Nits.

File: content/blog/2022/07/psp-migration-script.md
Comment (flavio) [reply content/blog/2022/07/psp-migration-script.md:8]: On top of that, I would be nice to make "past blog posts" link to older blog posts about this topic. If we have more than one, we could have "past" link to "post 1", "blog" link to "post 2" and so on so forth.

I don't know if we ever tagged blog posts, we could also have a "PSP" tag that collects all of them

File: content/blog/2022/07/psp-migration-script.md
Comment (flavio) [inline content/blog/2022/07/psp-migration-script.md:274]: I think it's worth add a note later on about the names used by the migration tool. We should point out that the names of the Kubewarden policies CR probably need to be tuned by the operators to be more human friendly

**Review by flavio — COMMENTED** [review]
That looks nice, I left some comments.

File: content/blog/2022/07/psp-migration-script.md
Comment (raulcabello) [inline content/blog/2022/07/psp-migration-script.md:272]:
```suggestion
kind: ClusterAdmissionPolicy
```

File: content/blog/2022/07/psp-migration-script.md
Comment (raulcabello) [inline content/blog/2022/07/psp-migration-script.md:96]: maybe we can mention also the --dry-run option?

File: content/blog/2022/07/psp-migration-script.md
Comment (viccuad) [reply content/blog/2022/07/psp-migration-script.md:22]: I will top it with

```suggestion
The script will migrate the PSPs defined in the `kubectl` context in use.
```

File: content/blog/2022/07/psp-migration-script.md
Comment (viccuad) [inline content/blog/2022/07/psp-migration-script.md:23]:
```suggestion
The Kubewarden policies will be printed to stdout. Thus, the users can
```

**Review by viccuad — COMMENTED** [review]
LGTM, modulo the other comments!

File: content/blog/2022/07/psp-migration-script.md
Comment (jvanz) [reply content/blog/2022/07/psp-migration-script.md:96]: In the latest version this options has been removed. Now, the script will print the policies in stdout. Thus, the user can redirect the policies definitions to `kubectl` or a file.

File: content/blog/2022/07/psp-migration-script.md
Comment (viccuad) [inline content/blog/2022/07/psp-migration-script.md:9]:
```suggestion
Kubewarden has 100% coverage of the deprecated, and soon removed,
```

**Review by viccuad — APPROVED** [review]
LGTM, there was a hiccup with the applied comments, but they are all applied. Just added 1 nitpick.

**Comment by jvanz** [comment]
@divya-mohan0209, I would like to get another round of review from you. If you approved, I'll updated the post date and merge it.

**Review by flavio — APPROVED** [review]

File: content/blog/2022/07/psp-migration-script.md
Comment (divya-mohan0209) [inline content/blog/2022/07/psp-migration-script.md:8]: We should consider putting this as a reference below instead of hyperlinking two words.

File: content/blog/2022/07/psp-migration-script.md
Comment (divya-mohan0209) [inline content/blog/2022/07/psp-migration-script.md:9]:
```suggestion
Kubewarden has 100% coverage of the deprecated, and soon to be removed,
```

File: content/blog/2022/07/psp-migration-script.md
Comment (divya-mohan0209) [inline content/blog/2022/07/psp-migration-script.md:14]: I think, we'd benefit from telling users, very briefly albeit, what AppVia's script does because we've modified it to suit our usecase.

File: content/blog/2022/07/psp-migration-script.md
Comment (divya-mohan0209) [reply content/blog/2022/07/psp-migration-script.md:18]: I don't think this has been added in the latest commit or maybe I'm not able to see the latest changes?

File: content/blog/2022/07/psp-migration-script.md
Comment (divya-mohan0209) [inline content/blog/2022/07/psp-migration-script.md:25]:
```suggestion
apply it directly or save it for further inspection. Let's take a look at an example:
```

**Review by divya-mohan0209 — CHANGES_REQUESTED** [review]
Few nits.

File: content/blog/2022/07/psp-migration-script.md
Comment (jvanz) [reply content/blog/2022/07/psp-migration-script.md]: I may rebased the branch an lost this change. Sorry.

File: content/blog/2022/07/psp-migration-script.md
Comment (jvanz) [reply content/blog/2022/07/psp-migration-script.md]: I just copy the same text from the docs. Let me know if you believe it is not enough.

File: content/blog/2022/07/psp-migration-script.md
Comment (divya-mohan0209) [inline content/blog/2022/07/psp-migration-script.md:14]:
```suggestion
automatically. The tool is capable of reading PSPs YAML and can generate the equivalent
```

File: content/blog/2022/07/psp-migration-script.md
Comment (divya-mohan0209) [inline content/blog/2022/07/psp-migration-script.md:15]:
```suggestion
policies in many different policy engines. Our simple script migrates
```

File: content/blog/2022/07/psp-migration-script.md
Comment (divya-mohan0209) [inline content/blog/2022/07/psp-migration-script.md:16]:
```suggestion
your PSPs to their equivalent Kubewarden policies.
```

File: content/blog/2022/07/psp-migration-script.md
Comment (divya-mohan0209) [inline content/blog/2022/07/psp-migration-script.md:21]:
```suggestion
the equivalent Kuberwarden policies' definitions in the standard output. Therefore,
```

File: content/blog/2022/07/psp-migration-script.md
Comment (divya-mohan0209) [inline content/blog/2022/07/psp-migration-script.md:28]: What does this PSP do? I'd like to reword it as
```suggestion
In a cluster with the PSP for <XYZ> applied:
```

File: content/blog/2022/07/psp-migration-script.md
Comment (divya-mohan0209) [inline content/blog/2022/07/psp-migration-script.md:98]:
```suggestion
the content to a file or review it directly on the console.
```

**Review by divya-mohan0209 — COMMENTED** [review]
Very minor nits. Rest LGTM! Thank you so much, Jose!

File: content/blog/2022/07/psp-migration-script.md
Comment (jvanz) [reply content/blog/2022/07/psp-migration-script.md:28]: It does a lot of things. It block access to host namespaces, block privileged containers, does not allow privilege escalation, forces the drop of all containers capabilities, list the allowed volume types, define the allowed user and groups to be used, controls the supplemental group applied to volumes and force containers to run in a read-only root filesystem.

File: content/blog/2022/07/psp-migration-script.md
Comment (jvanz) [reply content/blog/2022/07/psp-migration-script.md:28]: I'm addind this suggestion to the text and merging the PR. Thus, we can release this text Today, as expected. @divya-mohan0209 , give me your feedback here again when you have time. I'll update the blog post with the fixes

**Pull request merged**
